client/filestore-client/src/FileManager.ts

```diff
@@ -0,0 +1,82 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { uploadFile } from './CordovaFileUpload';
+import { FileQueue } from './FileQueue';
+import { uploadDataUrl } from './UriFileUpload';
+
+/**
+ * Manager for file uploads
+ */
+export class FileManager {
+  private uploadQueue;
+
+  public constructor(private baseUrl: string, private httpService, name: string) {
+    this.uploadQueue = new FileQueue(name);
+    // Start processing uploads on startup
+    this.startProcessingUploads();
+    // Listen when client becomes online for uploads
+    const self = this;
+    document.addEventListener('online', function() {
+      self.startProcessingUploads();
+    });
+  }
+
+  public startProcessingUploads() {
+    const queueItems = this.uploadQueue.restoreData().getItemList();
+    const self = this;
+    if (queueItems && queueItems.length > 0) {
+      console.info('Processing offline file queue. Number of items to save: ', queueItems.length);
+      let promise = this.saveFile(queueItems[0]).then(function() {
```

Comment:
The queue processing part might be the hardest part of the file feature...

I think the best function to deal with the concurrency for now would be
```typescript
Bluebird.map(queueItems, (i) => self.saveFile(i), {concurrency: 1});
```

See http://bluebirdjs.com/docs/api/promise.map.html#map-option-concurrency

Reply:
This one definitely need improvements. Great point!